### PR TITLE
Fixed: Update Shift Details Page for Mobile View Enhancements and Date Grouping (Issue #142)

### LINKED ISSUE #142: Update Shift Details Page for Mobile View Enhancements and Date Grouping
**Is your feature request related to a problem? Please describe.**
**Given** the current implementation of the "Shift Details" page, which displays all columns on mobile and uses "Send Emergency Alert" as a column name,
**When** users view the page on a mobile device or interact with the page,
**Then** they should experience improved usability, with only essential columns visible by default, clear labeling, and color-coded grouping of dates.

**Describe the solution you'd like**
Implement the following updates on the "Shift Details" page for the current sprint:

1) Mobile View Column Display:
On mobile, display only the Date, Shift, and Coverage columns by default. When users click on a specific shift row, expand the row to reveal the additional details such as View Volunteers, Open/Close Sign Up, Edit Shift, and Contact Volunteers columns.
2) Rename the "Send Emergency Alert" column:
Change the label of the "Send Emergency Alert" column to "Contact Volunteers" to better reflect its purpose.
3) Add Color Labels to Date Column:
Add small color labels to the left side of the Date column to visually group the shifts by date. Each date group should have a distinct color for better visual identification.
4) Styling Adjustments:
Make any necessary styling adjustments for consistency with the overall design and ensure that all changes remain responsive across devices and screen sizes.

**Acceptance Criteria**

- [ ] In mobile view, only the Date, Shift, and Coverage columns are displayed by default.
- [ ] Clicking on a specific shift row in mobile view expands the row to show additional details (i.e., View Volunteers, Open/Close Sign Up, Edit Shift, and Contact Volunteers).
- [ ] The "Send Emergency Alert" column is renamed to "Contact Volunteers" across the UI.
- [ ] Color labels are added to the left side of the Date column to group the shifts by date.
- [ ] All updates are responsive and display correctly across different screen sizes.
- [ ] Styling changes are consistent and match the overall design.

### PR DESCRIPTION
**Pull request by cbiddle3**
Fixes #142 

**What was changed?**

Changed the mobile view to only render 3  most important columns (Date, Shift, Coverage) and let the user expand the columns. The row is highlighted grey when the date differs from the previous ones (these will be in order when the table is rendered). Updated Emergency Alert to Contact Volunteers. 

**Why was it changed?**

Update to what our client wanted for the UI. 

**How was it changed?**

I had to update the css files from the Upcoming Shifts page modals because their styling affected all modals (I just added className attributes). I update the name and variable names to reflect contact volunteers. Use dynamic rendering and document queries to change the columns rendered based on the size of the window screen and in this file added buttons for collapsing and expanding those columns when the user is in mobile view. 

Screenshot 2024-11-10 at 7 39 01 PM
Screenshot 2024-11-10 at 7 39 13 PM
Screenshot 2024-11-10 at 7 39 18 PM
Screenshot 2024-11-10 at 7 39 23 PM

### CODE REVIEW

File: client_app/src/components/shelter/ShiftDetailsTable.js [hunk 1]
```diff
@@ -7,20 +7,33 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
...
+  const [areColsExpanded, setAreColsExpanded] = useState(false);
+  let prevStartDate = "0/0/0000";
+
+  useEffect(() => {
+    const handleResize = () => setIsMobileView(window.innerWidth < 430);
+    window.addEventListener("resize", handleResize);
+
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
```
Comment (amalrzv) [inline client_app/src/components/shelter/ShiftDetailsTable.js:33]: I think the mobile view could be more compressed if you removed more of the empty space with the titles and dates; shifts

File: client_app/src/components/shelter/ShiftDetailsTable.js [hunk 2]
```diff
@@ -93,32 +106,52 @@ export const ShiftDetailsTable = props => {
...
+      return "white";
+    }
+  }
+
   return (
     <div className="shiftdetails-container">
+      {(!areColsExpanded && isMobileView) && <button onClick={() => {setAreColsExpanded(true)}}>View All Columns</button>}
+      {(isMobileView && areColsExpanded) && <button onClick={() => {setAreColsExpanded(false)}}>Collapse Columns</button>}
       <table className="shiftsTable">
         <thead className="shiftsTableHeader" style={{position:stickyHeader,zIndex:headerZIndex,top:-1}}>
```
Comment (amalrzv) [inline client_app/src/components/shelter/ShiftDetailsTable.js:146]: I like the collapsable columns, its easy to view everything on the phone set up

File: client_app/src/components/shelter/ShiftDetailsTable.js (hunk 1 above)
Comment (cbiddle3) [reply client_app/src/components/shelter/ShiftDetailsTable.js:33]: Spoke with Amal she said it's okay then -- I asked if maybe we should not render name but she said no

File: client_app/src/components/shelter/ShiftDetailsTable.js (hunk 2 above)
Comment (cbiddle3) [reply client_app/src/components/shelter/ShiftDetailsTable.js:146]: She liked it was easy to see

**Review by Brijitha1609 — APPROVED** [review]
Excellent job! @cbiddle3. No issues were found during the review. The updates effectively align the UI with the client’s requirements, enhancing both functionality and user experience.

**Pull request merged**